js/gestionPresupuestoWeb.js

```diff
@@ -0,0 +1,302 @@
+    elemento.prepend(h1);
+    
+    for(let gasto in agrup){
+        let agrupa=auxCrearElemento('div','agrupacion-dato',"");
+        agrupa.prepend(auxCrearElemento('span','agrupacion-dato-clave',gasto[0]));
+        agrupa.prepend(auxCrearElemento('span','agrupacion-dato-valor',gasto[1]));
+        elemento.prepend(agrupa);
+
+    }
+    document.getElementById(idElemento).append(elemento);
+}
+
+function repintar(){
+    
+    let presupuesto = gp.mostrarPresupuesto();
+    mostrarDatoEnId("presupuesto", presupuesto);
+
+    let balanceT = gp.calcularBalance();
+    mostrarDatoEnId("balance-total",balanceT);
+
+    let gastosT = gp.calcularTotalGastos();
+    mostrarDatoEnId("gastos-totales",gastosT);
+
+    document.getElementById("listado-gastos-completo").innerHTML = "";
+    let gastoslistados = gp.listarGastos();
+
+    for (let gasto of gastoslistados){
+
+        mostrarGastoWeb("listado-gastos-completo",gasto)
+    }
+}
+
+function actualizarPresupuestoWeb(){
+
+    let nuevoPresupuesto = parseFloat(prompt("Introduce tu presupuesto, por favor"));
+    gp.actualizarPresupuesto(nuevoPresupuesto)
+    repintar();
+}
+
+
+function nuevoGastoWeb(){
+
+    let descripcionN = prompt("Introduce la descripción del gasto, por favor");
+
+    let valorN = prompt("Introduce el importe del gasto, por favor");
+    valorN = parseFloat(valorN);
+
+    let fechaN = prompt("Introduce la fecha del gasto con formato YYYY-MM-DD, por favor");
+
+    let etiquetasN = prompt("Introduce las etiquetas del gasto separadas por comas, por favor");
+    etiquetasN = etiquetasN.split(',');
+
+    let gasto = new gp.CrearGasto(descripcionN,valorN,fechaN,etiquetasN);
+    gp.anyadirGasto(gasto);
+
+    repintar();
+}
+
+
+function EditarHandle(){
+
+    this.handleEvent = function(e){
+
+        let descripcionNueva = prompt("Introduce una descripción del gasto, por favor : ",this.gasto.descripcion);
+        this.gasto.actualizarDescripcion(descripcionNueva);
+
+        let valorNuevo = prompt("Introduce el importe del gasto, por favor",this.gasto.valor);
+        valorNuevo = parseFloat(valorNuevo);
+        this.gasto.actualizarValor(valorNuevo);
+
+        let fechaNueva = prompt("Introduce la fecha del gasto con formato YYYY-MM-DD, por favor",this.gasto.fecha);
+        this.gasto.actualizarFecha(fechaNueva);
+
+        let etiquetasNuevas = prompt("Introduce las etiquetas del gasto separadas por comas, por favor",this.gasto.etiquetas);
+        etiquetasNuevas =etiquetasNuevas.split(',');
+        this.gasto.anyadirEtiquetas(...etiquetasNuevas);
+
+        repintar();
+    }
+}
+
+function BorrarHandle(){
+    this.handleEvent = function(e){
+
+        gp.borrarGasto(this.gasto.id);
+        repintar();
+    }
+}
+function BorrarEtiquetasHandle(){
+
+    this.handleEvent = function(e){
+
+        this.gasto.borrarEtiquetas(this.etiqueta);
+        repintar();
+    }
+}
+
+function nuevoGastoWebFormulario(){
+
+    this.handleEvent = function(e){
```

Comment:
El problema es que la función `nuevoGastoWebFormulario` no la usas como constructora, sino como función sencilla. Por tanto, no tienes que definir `this.handleEvent`: simplemente tienes que dejar el código de dentro.

Reply:
Gracias, ya me pasan los dos primeros test, pero los 2 últimos me dice lo mismo.